packages/caliper-core/test/worker/worker-message-handler.js

```diff
@@ -0,0 +1,171 @@
+        new WorkerMessageHandler(messengerMock, connectorFactoryMock);
+        connectorFactoryMock.resolves('connector-instance');
+        const message = {
+            getType: () => MessageTypes.Initialize,
+            stringify: () => 'initialize message'
+        };
+
+        const registeredHandler = messengerMock.on.getCalls().find(call => call.args[0] === MessageTypes.Initialize).args[1];
+        await registeredHandler(message);
+
+        sinon.assert.calledOnce(connectorFactoryMock);
+        sinon.assert.calledOnce(messengerMock.send);
+        sinon.assert.calledWith(messengerMock.send, sinon.match.instanceOf(ReadyMessage));
+    });
+
+    it('should prepare the test when receiving a "prepare" message', async () => {
+        const handler = new WorkerMessageHandler(messengerMock, connectorFactoryMock);
+        handler.worker = workerMock;
+        const message = {
+            getType: () => MessageTypes.Prepare,
+            getRoundIndex: () => 0,
+            stringify: () => 'prepare message'
+        };
+
+        workerMock.prepareTest.resolves();
+
+        const registeredHandler = messengerMock.on.getCalls().find(call => call.args[0] === MessageTypes.Prepare).args[1];
+        await registeredHandler(message);
+
+        sinon.assert.calledOnce(workerMock.prepareTest);
+        sinon.assert.calledOnce(messengerMock.send);
+        sinon.assert.calledWith(messengerMock.send, sinon.match.instanceOf(PreparedMessage));
+    });
+
+    it('should execute the test round when receiving a "test" message', async () => {
+        const handler = new WorkerMessageHandler(messengerMock, connectorFactoryMock);
+        handler.worker = workerMock;
+        const message = {
+            getType: () => MessageTypes.Test,
+            getRoundIndex: () => 0,
+            stringify: () => 'test message'
+        };
+
+        workerMock.executeRound.resolves();
+
+        const registeredHandler = messengerMock.on.getCalls().find(call => call.args[0] === MessageTypes.Test).args[1];
+        await registeredHandler(message);
+
+
+        sinon.assert.calledOnce(workerMock.executeRound);
+        sinon.assert.calledOnce(messengerMock.send);
+        sinon.assert.calledWith(messengerMock.send, sinon.match.instanceOf(TestResultMessage));
+    });
+
+    it('should resolve the exit promise when receiving an "exit" message', async () => {
+        const handler = new WorkerMessageHandler(messengerMock, connectorFactoryMock);
+        handler.exitPromiseFunctions.resolve = sandbox.stub();
+        const message = {
+            getType: () => MessageTypes.Exit,
+            stringify: () => 'exit message'
+        };
+
+
+        const exitHandler = messengerMock.on.getCalls().find(call => call.args[0] === MessageTypes.Exit).args[1];
+        await exitHandler(message);
+
+        handler.exitPromiseFunctions.resolve.should.have.been.calledOnce;
+    });
+
+    it('should throw Error when constructor validations are violated', () => {
```

Comment:
This should be in it's own describe "When creating a Worker Message Handler"
We should also split this into 3 tests because we want to provide details of each failure, the above doesn't tell me what the constructor validations are